python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+    """
+    Parameters
+    ----------
+    fpx, fpy : array
+        Focal plane position in millimeters in DVCS
+        See https://lse-349.lsst.io/
+    det : lsst.afw.cameraGeom.Detector
+        Detector of interest.
+
+    Returns
+    -------
+    x, y : array
+        Pixel coordinates.
+    """
+    tx = det.getTransform(cameraGeom.FOCAL_PLANE, cameraGeom.PIXELS)
+    x, y = tx.getMapping().applyForward(np.vstack((fpx, fpy)))
+    return x.ravel(), y.ravel()
+
+
+def pixel_to_focal(x, y, det):
+    """
+    Parameters
+    ----------
+    x, y : array
+        Pixel coordinates.
+    det : lsst.afw.cameraGeom.Detector
+        Detector of interest.
+
+    Returns
+    -------
+    fpx, fpy : array
+        Focal plane position in millimeters in DVCS
+        See https://lse-349.lsst.io/
+    """
+    tx = det.getTransform(cameraGeom.PIXELS, cameraGeom.FOCAL_PLANE)
+    fpx, fpy = tx.getMapping().applyForward(np.vstack((x, y)))
+    return fpx.ravel(), fpy.ravel()
+
+
+def stamp_to_ccd(stamp, ccdimg, detector, camera, ampName, ampCol, ampRow):
+    """
+    Place ROI (stamp) into existing full CCD array, all in CCD view
+    (Should rewrite to use mk_roi_bboxes(md,camera))
+
+    Parameters
+    ----------
+    stamp : array
+        raw ROI array in amp coordinates
+    ccdimg : array
+        ccd image array in ccd coordinates
+    detector : lsst.afw.cameraGeom.Detector
+        Detector of interest.
+    camera : lsst.afw.cameraGeom.Camera
+        Camera object
+    ampName : str
+        amplifier name, eg. C00
+    ampCol : int
+        starting column number for ROI
+    ampRow : int
+        starting row number for ROI
+
+    Returns
+    -------
+    ccdimg : array
+        ccd image array filled with ROI
+    """
+
+    stamp_ccd = amp_to_ccdview(stamp, detector, ampName)
+    ampRows, ampCols = stamp_ccd.shape
+
+    # get corner0 of the location for the ROI
+    lct = LsstCameraTransforms(camera, detector.getName())
+    corner0_CCDX, corner0_CCDY = lct.ampPixelToCcdPixel(ampCol, ampRow, ampName)
+    corner0_CCDX = int(corner0_CCDX)
+    corner0_CCDY = int(corner0_CCDY)
+
+    # get opposite corner, corner2, of the ROI
+    amp = detector[ampName]
+    if amp.getRawFlipX():
+        corner2_CCDX = corner0_CCDX - ampCols
+    else:
+        corner2_CCDX = corner0_CCDX + ampCols
+
+    if amp.getRawFlipY():
+        corner2_CCDY = corner0_CCDY - ampRows
+    else:
+        corner2_CCDY = corner0_CCDY + ampRows
+
+    # now place the ROI
+    try:
+        ccdimg[
+            min(corner0_CCDY, corner2_CCDY) : max(corner0_CCDY, corner2_CCDY),
+            min(corner0_CCDX, corner2_CCDX) : max(corner0_CCDX, corner2_CCDX),
+        ] = stamp_ccd
+    except Exception:
+        print(detector.getName(), ampName)
+        print("Y: ", min(corner0_CCDY, corner2_CCDY), max(corner0_CCDY, corner2_CCDY))
+        print("X: ", min(corner0_CCDX, corner2_CCDX), max(corner0_CCDX, corner2_CCDX))
```

Comment:
These should be log.warnings and this should probably raise too, after sending the messages, not just return